File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.java
class Solution {
    public int[] intersect(int[] nums1, int[] nums2) {
        int i,j;
        List<Integer> a=new ArrayList<Integer>();
        Arrays.sort(nums1);
        Arrays.sort(nums2);
        i=0;j=0;
        while(i<nums1.length && j<nums2.length){
            if(nums1[i]<nums2[j]){
                i++;
            }
            else if(nums1[i]>nums2[j]){
                j++;
            }
            else{
                a.add(nums1[i]);
                i++;j++;
            }
        }
        int[] arr = a.stream().mapToInt(k-> k).toArray();
        return arr;
    }
}

File: 0350-intersection-of-two-arrays-ii/0350-intersection-of-two-arrays-ii.py
class Solution:
    def intersect(self, n1: List[int], n2: List[int]) -> List[int]:
        a=[]
        nums1=sorted(n1)
        nums2=sorted(n2)
        i=0
        j=0
        while(i<len(nums1) and j<len(nums2)):
            if(nums1[i]>nums2[j]):
                j=j+1
            elif(nums1[i]<nums2[j]):
                i=i+1
            else:
                a.append(nums1[i])
                i=i+1
                j=j+1
        return a